src/Microsoft.ML.StandardLearners/Standard/Online/OnlineLinear.cs

```diff
@@ -16,30 +16,49 @@
 
 namespace Microsoft.ML.Trainers.Online
 {
-
+    /// <summary>
+    /// Arguments class for online linear trainers.
+    /// </summary>
     public abstract class OnlineLinearArguments : LearnerInputBaseWithLabel
```

Comment:
>OnlineLinearArguments [](start = 26, length = 21)

think those are supposed to be called Options. cc @artidoro, @abgoswam

Reply:
Not really. This serves as the base class  (for the  online linear learners) from which the public facing `Options` class  is derived.

It should be `internal` though.  I believe Issue #2264 should take care of making it internal

Reply:
I would like to disagree regarding `internal`. Why? If you make it internal all this property will disappear for user, and they are good properties.
What is wrong with having hierarchy of classes?

---
In reply to: [256639493](https://github.com/dotnet/machinelearning/pull/2517#discussion_r256639493) [](ancestors = 256639493)

Reply:
I was going by _smell_ test -- I do not see any samples (yet) of how this class would be helpful for a public facing API.  I may be wrong though.  

Would like to hear @yaeldekel / @codemzs thoughts on this since they are working on "lockdown" issues. 

If we do want to keep this `public` perhaps we should rename to OnlineLinear**Options** ?

Reply:
We have following structure:
OnlineLinear Options -> LinearSVM Options
|
|____> AverageLinear Options--> AveragePerceptron Options
                  |
                  |____________> OnlineGradientDescent Options
base options contains properties which belong to all of them, derived classes contains specific options.

Only way how you can get rid of this tree structure is to create new flatten object which will be flat and have all required properties by himself. But that would also mean you need to repeat some properties in this flatten structures multiple times.

---
In reply to: [256646030](https://github.com/dotnet/machinelearning/pull/2517#discussion_r256646030) [](ancestors = 256646030)